phpBB/phpbb/notification/manager.php

```diff
@@ -910,18 +917,30 @@ public function get_item_type_class($notification_type_name, $data = array())
 	 * Helper to get the notifications method class and set it up
 	 *
 	 * @param string $method_name
+	 *
 	 * @return method\method_interface
+	 * @throws \Exception When object name is not o notification method
 	 */
 	public function get_method_class($method_name)
 	{
-		return $this->load_object($method_name);
+		$object = $this->load_object($method_name);
+
+		if (!$object instanceof method\method_interface)
+		{
+			throw new \Exception('Supplied method name returned invalid service: ' . $method_name);
```

Comment:
I would use runtime_exception from phpbb\exception, or at least RuntimeException for all of this exceptions